development/gcp/cloud-functions/rotateserviceaccount/rotateserviceaccount.go

```diff
@@ -0,0 +1,126 @@
+	"github.com/kyma-project/test-infra/development/gcp/pkg/pubsub"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/secretmanager"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/secretversionsmanager"
+	"google.golang.org/api/iam/v1"
+)
+
+var (
+	projectID                   string
+	secretManagerService        *secretmanager.Service
+	secretVersionManagerService *secretversionsmanager.Service
+	serviceAccountService       *iam.Service
+)
+
+// ServiceAccountJSON stores Service Account athentication data
+type ServiceAccountJSON struct {
+	Type             string `json:"type"`
+	ProjectID        string `json:"project_id"`
+	PrivatekayID     string `json:"private_key_id"`
+	PrivateKey       string `json:"private_key"`
+	ClientEmail      string `json:"client_email"`
+	ClientID         string `json:"client_id"`
+	AuthURL          string `json:"auth_uri"`
+	TokenURI         string `json:"token_uri"`
+	AuthProviderCert string `json:"auth_provider_x509_cert_url"`
+	ClientCert       string `json:"client_x509_cert_url"`
+}
+
+func init() {
+	var err error
+	ctx := context.Background()
+
+	projectID = os.Getenv("GCP_PROJECT_ID")
+
+	secretManagerService, err = secretmanager.NewService(ctx)
+	if err != nil {
+		panic(fmt.Sprintf("failed creating Secret Manager client, error: %s", err.Error()))
+	}
+	secretVersionManagerService = secretversionsmanager.NewService(secretManagerService)
+
+	serviceAccountService, err = iam.NewService(ctx)
+	if err != nil {
+		panic(fmt.Sprintf("failed creating IAM client, error: %s", err.Error()))
+	}
+}
+
+// RotateServiceAccount adds new secret version in Secret Manager on pubsub rotate message
+func RotateServiceAccount(ctx context.Context, m pubsub.MessagePayload) error {
+	var err error
+	var secretRotateMessage pubsub.SecretRotateMessage
+	var secretData ServiceAccountJSON
+
+	// Create logger to use google cloud functions structured logging
+	logger := cloudfunctions.NewLogger()
+	// Set component for log entries to identify all messages for this function.
+	logger.WithComponent("kyma.prow.cloud-function.RotateServiceAccount")
+	// Set trace value for log entries to identify messages from one function call.
+	logger.GenerateTraceValue(projectID, "RotateServiceAccount")
+
+	if m.Attributes["eventType"] != "SECRET_ROTATE" {
+		logger.LogInfo(fmt.Sprintf("Unsupported event type: %s, quitting\n", m.Attributes["eventType"]))
+		return nil
+	}
+
+	if secretRotateMessage.Labels["type"] != "service-account" {
+		logger.LogInfo(fmt.Sprintf("Unsupported secret type: %s, quitting\n", secretRotateMessage.Labels["type"]))
+		return nil
+	}
+
+	err = json.Unmarshal(m.Data, &secretRotateMessage)
+	if err != nil {
+		logger.LogCritical(fmt.Sprintf("failed to unmarshal message data field, error: %s", err.Error()))
+	}
+
+	//get latest secret version data
+	secretlatestVersionPath := secretRotateMessage.Name + "/versions/latest"
+	logger.LogInfo(fmt.Sprintf("Retrieving %s secret", secretlatestVersionPath))
```

Comment:
It's more of a nitpick, but you don't need to use `fmt.Sprintf` here. Just type the code below:
```suggestion
	logger.LogInfo("Retrieving secret: "+secretlatestVersionPath)
```